### PR TITLE
make the coprocessor URL configurable per stage

### PR DESCRIPTION
**Pull request by Geal**
This makes it possible to have different coprocessor URLs per stage. The initial URL configuration still serves as a global default, but it can be overriden per stage

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: apollo-router/src/plugins/coprocessor/supergraph.rs [hunk 1]
```diff
@@ -53,6 +55,8 @@ pub(super) struct SupergraphResponseConf {
     pub(super) status_code: bool,
     /// Handles the response without waiting for the coprocessor to respond
     pub(super) detached: bool,
+    /// The url you'd like to offload processing to
+    pub(super) url: Option<String>,
```
Comment (bnjjj) [inline apollo-router/src/plugins/coprocessor/supergraph.rs:59]:
```suggestion
    #[schemars(with = "String")]
    pub(super) url: Option<url::Url>,
```

Comment (bnjjj) [reply apollo-router/src/plugins/coprocessor/supergraph.rs:59]: To benefit from url check when deserializing

**Review by bnjjj — COMMENTED** [review]
Missing documentation updates

File: apollo-router/src/plugins/coprocessor/supergraph.rs (hunk 1 above)
Comment (Geal) [reply apollo-router/src/plugins/coprocessor/supergraph.rs:59]: oh nice

File: .changesets/feat_geal_coprocessor_urls.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+```yaml title="router.yaml"
+coprocessor:
+  url: http://127.0.0.1:8081
+  router:
+    request:
+      headers: true
+  supergraph::
+    request:
+      body; true
```
Comment (o0Ignition0o) [inline .changesets/feat_geal_coprocessor_urls.md:15]:
```suggestion
      body: true
```

**Review by o0Ignition0o — APPROVED** [review]

**Pull request merged**